### ISSUE TITLE
Allow audits to be run on local (or linked) modules

### ISSUE THREAD

**Issue by radiovisual** [npm/npm#21155]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] npm is producing incorrect or undesirable behavior.
  - [x] Other (_see below for feature requests_):

#### What's going wrong?

`npm audit` cannot be run on my local source files. If a problem is discovered in one of your published modules, and you go to fix the problem, there is no way to run the audit again in order to validate that the change you made was enough to fix the problem.

#### What's the feature?

Letting us pass file(s) into the audit script, something like: 

- `npm audit local` Runs an audit agaignst all files in the `files` array in the package.json file. or,
- `npm audit --file=/path/to/file` would run an audit against the provided file

#### What problem is the feature intended to solve?

This will aid in the development/debugging/valiadating fixes to reported security vulnerabilities

#### Is the absence of this feature blocking you or your team? If so, how?

In a way, yes, because there is not a straight-forward way to test that local changes to source files is enough to pass the security checks. 

#### Is this a feature you're prepared to implement, with support from the npm CLI team?

Maybe! ™️ 🍻

**Comment by christian-bromann** [comment]
I think this issue is based on the confusion that people think NPM is running actual security checks on code when it only looks into the Nodesecurity database if there are any entries made for the project. It would be nice to be more transparent about this and allow package owner to ask about a new security review after they have patched the vulnerability (by updating an old dependency or patching an actual security violation).

**Comment by radiovisual** [comment]
> I think this issue is based on the confusion that people think NPM is running actual security checks on code when it only looks into the Nodesecurity database if there are any entries made for the project.

Yup, I am guilty of this misunderstanding to some extent: I thought that there was a real security check, and then the Nodesecurity database was updated, and I was hoping that the updates were dynamic in the sense that if the security audit passes after an initial failure, then the old security report would be marked as "closed" or "resolved".

I agree that this system is not very flexible if it requires a manual step of having to hunt down the original reporter and ask them to manually update.
